pkg/observability/metrics.go

```diff
@@ -0,0 +1,48 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package observability
+
+import (
+	"github.com/kubearchive/kubearchive/pkg/database/interfaces"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/metric"
+)
+
+var CloudEvents metric.Int64UpDownCounter
+
+type CEResult string
+
+const (
+	CEResultInsert  CEResult = "insert"
+	CEResultUpdate  CEResult = "update"
+	CEResultNone    CEResult = "none"
+	CEResultError   CEResult = "error"
+	CEResultNoMatch CEResult = "no_match"
+)
+
+func NewCEResultFromWriteResourceResult(result interfaces.WriteResourceResult) CEResult {
+	if result == interfaces.WriteResourceResultInserted {
+		return CEResultInsert
+	} else if result == interfaces.WriteResourceResultUpdated {
+		return CEResultUpdate
+	} else if result == interfaces.WriteResourceResultNone {
+		return CEResultNone
+	} else if result == interfaces.WriteResourceResultError {
+		return CEResultError
+	}
+
+	return CEResultError
```

Comment:
I think a switch fits better here:

```suggestion
    switch result {
        case interfaces.WriteResourceResultInserted:
            return CEResultInsert
        case interfaces.WriteResourceResultUpdated:
            return CEResultUpdate
        case interfaces.WriteResourceResultNone:
            return CEResultNone
        case interfaces.WriteResourceResultError:
            return CEResultError
        default:
            return CEResultError
    }
```

Reply:
Changed to switch